processor/k8sattributesprocessor/processor.go

```diff
@@ -151,25 +151,41 @@ func (kp *kubernetesprocessor) processResource(ctx context.Context, resource pco
 // addContainerAttributes looks if pod has any container identifiers and adds additional container attributes
 func (kp *kubernetesprocessor) addContainerAttributes(attrs pcommon.Map, pod *kube.Pod) {
 	containerName := stringAttributeFromMap(attrs, conventions.AttributeK8SContainerName)
-	if containerName == "" {
+	containerID := stringAttributeFromMap(attrs, conventions.AttributeContainerID)
+	var (
+		containerSpec *kube.Container
+		ok            bool
+	)
+	switch {
+	case containerName != "":
+		containerSpec, ok = pod.Containers.ByName[containerName]
+		if !ok {
+			return
+		}
+	case containerID != "":
+		containerSpec, ok = pod.Containers.ByID[containerID]
+		if !ok {
+			return
+		}
+	default:
 		return
 	}
-	containerSpec, ok := pod.Containers[containerName]
-	if !ok {
-		return
+	if containerSpec.Name != "" {
+		if _, ok := attrs.Get(conventions.AttributeK8SContainerName); !ok {
+			attrs.PutStr(conventions.AttributeK8SContainerName, containerSpec.Name)
+		}
 	}
-
 	if containerSpec.ImageName != "" {
-		if _, found := attrs.Get(conventions.AttributeContainerImageName); !found {
+		if _, ok := attrs.Get(conventions.AttributeContainerImageName); !ok {
```

Comment:
nit: please avoid unrelated changes in the diff

Reply:
Ah, this is a habit of mine that doesn't apply on this repository... Sorry, fixed.